src/win/pipe.c

```diff
@@ -1246,7 +1247,12 @@ static int uv_pipe_write_impl(uv_loop_t* loop,
     if (send_handle) {
       tcp_send_handle = (uv_tcp_t*)send_handle;
 
-      err = uv_tcp_duplicate_socket(tcp_send_handle, handle->pipe.conn.ipc_pid,
+      pid = handle->pipe.conn.ipc_pid;
+      if (pid == 0) {
+        pid = uv_current_pid();
+      }
```

Comment:
:astonished: Really? The file was already packed with braces around single statement if's.